orchestrator/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,480 @@
+package kubernetes
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+
+	"github.com/rancher/longhorn-manager/crdstore"
+	"github.com/rancher/longhorn-manager/orchestrator"
+	"github.com/rancher/longhorn-manager/types"
+	"github.com/rancher/longhorn-manager/util"
+
+	apiv1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kCli "k8s.io/client-go/kubernetes"
+)
+
+const (
+	cfgDirectory = "/var/lib/rancher/longhorn/"
+	hostUUIDFile = cfgDirectory + ".physical_host_uuid"
+)
+
+var (
+	WaitDeviceTimeout = 30 //seconds
+	WaitAPITimeout    = 30 //seconds
+	WaitPodTimeout    = 5  //seconds
+	WaitPodPeriod     = 3  //seconds
+	WaitPodCounter    = 5
+)
+
+type Kubernetes struct {
+	EngineImage string
+	IP          string
+	currentNode *types.NodeInfo
+	cli         *kCli.Clientset
+}
+
+type Config struct {
+	EngineImage string
+}
+
+func NewKuberOrchestrator(cfg *Config) (*Kubernetes, error) {
+	if cfg.EngineImage == "" {
+		return nil, fmt.Errorf("missing required parameter EngineImage")
+	}
+
+	config, err := crdstore.GetClientConfig("")
+	if err != nil {
+		return nil, err
+	}
+
+	kube := &Kubernetes{
+		EngineImage: cfg.EngineImage,
+	}
+
+	kube.cli, err = kCli.NewForConfig(config)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot connect to kubernetes")
+	}
+
+	if err = kube.updateNetwork(); err != nil {
+		return nil, errors.Wrapf(err, "fail to detect dedicated pod network")
+	}
+
+	logrus.Infof("Detected  IP is %s", kube.IP)
+
+	if err := kube.updateCurrentNode(); err != nil {
+		return nil, err
+	}
+
+	logrus.Info("Kubernetes orchestrator is ready")
+	return kube, nil
+}
+
+func (k *Kubernetes) updateNetwork() error {
+	IP := os.Getenv("POD_IP")
```

Comment:
As I said, get all the necessary envs at the initialization and store the result in the object. Don't call `os.Getenv` everytime when you need it.